inst/tests/tests.Rraw

```diff
@@ -21079,3 +21079,8 @@ setindex(DT, b)
 # make sure that print(DT) doesn't warn due to the header missing index column types, #6806
 # can't use output= here because the print() call is outside withCallingHandlers(...)
 test(2307, { capture.output(print(DT, class = TRUE, show.indices = TRUE)); TRUE })
+
+# fread with colClasses and keepLeadingZeros=TRUE #6851
+dt = data.table(date=as.IDate(c(NA, "2014-12-05")))
+test(2308.01, fread("date\nNA\n2014-12-05", keepLeadingZeros=TRUE), dt)
```

Comment:
Interesting that this test request fails the same way in 6c69c15801474eb5a550c6f3233e50b9a835002f (immediately after `1.16.0`) as in `master`, but the problem was masked for the example from #6851 until #6107.

Reply:
Do you think 'regression' is still the right way to categorize this issue?

BTW, this test and the NEWS bullet don't add up for me -- neither of these tests have any leading zeroes.

Maybe the bullet needs some refinement, or perhaps an example?

Reply:
They have leading zeros since year, month and date are parsed separately.

This is also the reason why the first date in the bad.csv in the issue isn't triggering it